structure/fenwicktree/fenwicktree.go

```diff
@@ -0,0 +1,62 @@
+//Fenwick Tree Data Structure for Range Queries. Also known as Binary Indexed Tree
+//Build: O(N)
+//Query: O(log(N))
+//Update: O(log(N))
+//reference: https://brilliant.org/wiki/fenwick-tree/
+
+package fenwicktree
+
+// n: Size of the input array.
+// array: the input array on which queries are made.
+// bit: store the sum of ranges.
+type FenwickTree struct {
+	n     int
+	array []int
+	bit   []int
+}
+
+// This function creates a new Fenwick tree, initialized array and bit with
+// the values of the array. Note that the queries and updates should have
+// one based indexing.
+func NewFenwickTree(array []int) *FenwickTree {
+	n := len(array)
+	newArray := []int{0} // Appending a 0 to the beginning as this implementation uses 1 based indexing
+	fenwickTree := &FenwickTree{
+		n:     n,
+		array: append(newArray, array...),
```

Comment:
A `copy` makes more sense here.

Reply:
Okay, will change

Reply:
Do you mean creating a copy of the array ?

Reply:
The array: https://go.dev/ref/spec#Appending_and_copying_slices

Reply:
But why do we need to use copy ? It might make the code more verbose

Reply:
Could you please elaborate a little more on why using copy is better instead of append ?

Reply:
because thats what you are doing? copying a slice?

Reply:
Not exactly, because we still need to append a 0 to the beginning and using copy would require to declare a variable first to copy to the destination variable, so I guess using append would be fine. What do you think about it ?

Reply:
I guess its not that big of an issue.

Reply:
So, is append fine or should I remove it and use copy?

Reply:
Append is fine.

Reply:
okay